tensorboard/data/server/server.rs

```diff
@@ -286,6 +425,36 @@ mod tests {
         );
     }
 
+    /// Converts a list-of-lists back into a map, for easy assertions that don't depend on
```

Comment:
Heh, I'll take a "clever" :) Yeah, I realize it depends on the orphan rule; originally I was thinking it would just be a new test-only trait with `fn get()` but then I realized `Index` could be used and makes it even pithier.

But I think the macro version is quite nice and addresses the main things I was lukewarm about with `unroll_map` so I'm perfectly happy to keep it the way you have it now.

Reply:
Yeah, I would also be totally fine with a test-only extension trait (no
orphan rule issues since the trait is in-crate). And I would probably
implement those with a macro.

I still might switch to that. It has type information earlier, so it’s
easier to return a `HashMap<Run, HashMap<Tag, V>>` given a mapping
function `FnMut(Self::TagEntry) -> V`. And that would be nice to avoid
the newly introduced `.as_ref()` dances. You can do it with the macro,
but it’s a bit annoying because of inference details. (Basically: you
can’t write `(|x| x.foo())(some_x)` without explicitly specifying the
type for `x`, but you can work around this by capturing `x`’s type, Java
style: `std::iter::once(some_x).map(|x| x.foo()).nth(0).unwrap()` is
one awkward way to do so.)